File: python/strategy/ActionGame/client/Character.py
from behavior.WeaponBehavior import WeaponBehavior

class Character:

    def __init__(self):
        self._weapon_behavior = WeaponBehavior

    def fight(self):
        self._weapon_behavior.use_weapon()

    def set_weapon(self, wb):
        self._weapon_behavior = wb

File: javascript/strategy/ActionGame/client/Character.js
import {WeaponBehavior} from "../behavior/WeaponBehavior.js";

export class Character {

    constructor() {
        this.weapon_behavior = new WeaponBehavior()
    }

    fight() {
        this.weapon_behavior.useWeapon()
    }

    setWeapon(wb) {
        this.weapon_behavior = wb
    }
}